---
title: "main"
author: "lijunx"
date: "May 30, 2023"
output: html_document
---

```{r getPackageFunction, echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("purrr")
library("data.table")
library("tidyverse")
source("scripts/supporting_function.R")

```

```{bash setup directory}
mkdir -p data/supporting
mkdir -p data/processed
mkdir -p data/output_error
mkdir -p data/processed/methylation
mkdir -p data/processed/mRNA
mkdir -p data/processed/copynumber
mkdir -p data/processed/VCF
mkdir -p data/processed/Signatures
mkdir -p data/processed/Signatures/tables
mkdir -p data/processed/Signatures/figures
mkdir -p data/processed/msi_icgc
mkdir -p data/processed/hrd_icgc
mkdir -p data/processed/hrd_icgc/WGS_HRD_score
mkdir -p data/processed/hrd_icgc/Array_HRD_score
mkdir plots
mkdir -p plots/methylation
mkdir -p plots/methylation/HR_gene_line
mkdir -p plots/mRNA
mkdir -p plots/mRNA/QC

```

```{r setup working directory}
dir <- getwd()
Sys.setenv(dir = dir)
```

# Methylation preprocessing
```{r preprocess methylation sample info}
# generate essential file for methylation process - TCGA
pd_file_list <- Sys.glob("data/raw/*/*[!maf]_sample.tsv")
invisible(lapply(pd_file_list, tcga_pd_generator))

#generate essential file for methylation process - ICGC
icgc_pd <- geo_pd_generator("data/raw/20220505_ICGC_OV_meth/GSE65820_series_matrix.txt", "!Sample_source_name_ch1")
icgc_pd_id <- cbind(Sample_Name = unlist(fread("data/raw/20220505_ICGC_OV_meth/GSE65820_series_matrix.txt", header = F, sep = "\t", skip = "ID_REF", nrows = 1)[,-1]),
                    Sample.ID= unlist(fread("data/raw/20220505_ICGC_OV_meth/GSE65820_series_matrix.txt", header = F, sep = "\t", skip = "Sample_title", nrows = 1 )[,-1])) %>%
  as.data.frame()
##manual currate pd before save
icgc_pd <- icgc_pd %>%
  left_join(icgc_pd_id) %>%
  mutate(Sample.ID_merge = gsub("_","", Sample.ID)) %>%
  
  left_join(fread("data/raw/20220505_ICGC_OV_meth/icgc_sample_id_mapping.tsv") %>%
             mutate(Sample.ID_merge = gsub("_","", paste0(`Case ID`, `Tumour sample ID`)))%>%
              dplyr::select(Sample.ID_merge, Sample_time = `Sample time point`, Tissue_type = `Sample type`)) %>%
  mutate(Sample_Group = case_when(str_detect(Sample_time, "relapse") ~"Relapse",
                                  str_detect(Sample_time, "metastasis") ~"Metastasis",
                                  str_detect(Sample_Group, "fallopian") ~"Solid Tissue Normal",
                                  str_detect(Sample_time, "(?i)primary") ~ "Primary Tumour")) %>%
  dplyr::select(-Sample.ID_merge, -Sample_time)
write.csv(icgc_pd, "data/raw/20220505_ICGC_OV_meth/IDAT/ICGC_OV_pd.csv", row.names = FALSE)

Sys.glob("data/raw/*")%>%
  as.data.frame() %>%
  `colnames<-` (c("IDAT")) %>%
  mutate(Project = gsub("data/raw/20220401_|data/raw/20220505_|_meth/IDAT/", "", IDAT)) %>%
  dplyr::select(Project, IDAT) %>%
  write.table("data/supporting/idat_mapping.tsv", row.names = F, quote = F, col.names = F)
  
```

```{bash methylation calling}
#obtain normalised beta
bash scripts/1_run_beta_norm.sh data/supporting/idat_mapping.tsv

#obtain p-value from detection
bash scripts/1_run_beta_pvalue.sh data/supporting/idat_mapping.tsv
```

```{bash extract QC information}
# monitor for job finished then proceed
running=$(wc -l data/supporting/idat_mapping.tsv | cut -d ' ' -f 1)
finished=$(ls | grep -E "beta_pval_*.o*" | wc -l)
while [ $finished != $(($running * 2)) ]; do running=$(wc -l data/supporting/idat_mapping.tsv | cut -d ' ' -f 1); finished=$(ls | grep -E "beta_pval_*.o*" | wc -l); sleep 20; done


#extract pvalue for duplicate filtration
for a in beta_pval*_meth.o*;
  do
    project=$(echo $a| cut -d '_' -f 3,4);
    filename=$(echo $project"_beta_pvalue.csv");
    tail -n +3 $a  > data/processed/methylation/$filename;
    done

#move all the pbs output
mv norm_beta*.* data/output_error/
mv beta_pval*.* data/output_error/
```

## Methylation visualisation
the normalised beta matrix for methylation will be deduplicated based on p-value for intensity drop-outs
beta matrix will be subset focusing just around the HR gene region, the gene include BRCA1, BRCA2, RAD51C, RAD51D, BRIP1, PALB2, CHECK2
CpG will be annotated based on gene promoter
line plots will be generated for genome context
```{bash methylation assessment for HR gene}
bash scripts/1_beta_visualisation.sh data/supporting/idat_mapping.tsv

#clean output
running=$(wc -l data/supporting/idat_mapping.tsv | cut -d ' ' -f 1)
finished=$(ls | grep -E "beta_plot_*.o*" | wc -l)
while [ $finished != $(($running * 2)) ]; do running=$(wc -l data/supporting/idat_mapping.tsv | cut -d ' ' -f 1); finished=$(ls | grep -E "beta_plot_*.o*" | wc -l); sleep 20; done
rm -rf beta_plot_*.*
```

# mRNA preprocessing
```{bash mRNA calling}
bash scripts/2_run_mRNA_edgeR.sh data/supporting/idat_mapping.tsv

# clean directory
running=$(qstat | grep "mRNA_" |wc -l)
while [ $running != 0 ]; do running=$(qstat | grep "mRNA_" |wc -l); sleep 20; done

# extract DEG counts for group contrast as well as whether silencing effect observed from DEG level
bash scripts/2_extractDEG.sh
mv mRNA* data/output_error/
```
# VEP
```{bash vep annotation of the somatic mutation}
bash scripts/3_run_vep.sh $dir
#bash scripts/3_run_vep_depmap.sh $dir
```
# Mutation Signature 
```{bash}
bash scripts/4_run_cosmic_signature.sh $dir
```

# MSI ICGC 
```{bash msi sensor for ICGC ovarian}
#bash scripts/5_run_msisensor.sh
```

# HRD score ICGC
please run scripts/6_HRDscore_ICGC_OV.Rmd

# Methylation clustering that is equivalent to Yates et al 
please run scripts/7_methylation_clustering ICGC_OV.Rmd

# gene level Copy number
```{bash}
for i in data/raw/*/copynumber; 
  do
    project=${i%%+(_meth/copynumber)};
    project=${project##*20220401_};
    echo $project;
    grep -Er "BRCA1|RAD51C" $i | grep -v "BRCA1P1" |  sed 's/^\.\///g' > data/processed/copynumber/${project}_BRCA1_RAD51C_cnv.tsv;
  done
```