---
title: "Mutational Signature analysis and HRD prediction"
author: "Olga Kondrashova"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
---


```{r getPackageFunction, echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
# Load required packages
library(tidyverse)
library(magrittr)
library(MutationalPatterns)
library(deconstructSigs)
library(scarHRD)
library(YAPSA)
library(here)
```

```{r setup, include=FALSE}
# knitr global options
knitr::opts_chunk$set( comment = NA, cache=TRUE, message = FALSE, warnings = FALSE, include = TRUE, fig.align = 'center', figcap.prefix = "Figure", figcap.sep = ":", figcap.prefix.highlight = "**", fig.path = "figures/")

options(scipen = 999)
```

### Preparation

Define functions for running mutational signatures & getting HRdetect scores. 

```{r define_functions}
run_deconstructsigs <- function(data,out_path, out_name, signatures, method = "default", plot=FALSE, cutoff=0.10) {
  # Run deconstructSigs on all samples
  for (i in 1:nrow(data)){
    sample_id=row.names(data[i,])
    sigs= whichSignatures(tumor.ref = data, 
                          signatures.ref = signatures, 
                          sample.id = sample_id,
                          signature.cutoff = cutoff,
                          contexts.needed = TRUE,
                          tri.counts.method = method)
    # Plot signatures if defined
    if ( isTRUE(plot)){
      out_file <- paste0(out_path,"figures/",sample_id,"_",method,"_",out_name,".pdf")
      pdf(out_file)
      plotSignatures(sigs)
      dev.off()
    }
    # Combine sig assignments for samples to one dataframe
    if (exists("sigs_final")) { 
      sigs_final <- rbind(sigs_final,sigs$weights) 
    } else { 
      sigs_final <- sigs$weights
    }
  }
  # Subset only detected signatures
  sigs_detected <- sigs_final %>%
    rownames_to_column(var="Sample")%>%
    gather(Sig, Count, -Sample) %>%
    group_by(Sig) %>%
    filter(Count > 0) %>%
    spread(Sig, Count, fill = 0)
  
  out_file <- paste0(out_path,"tables/",method,"_",out_name,".tsv")
  # Return signature assigment dataframe
  write.table(sigs_detected, out_file ,quote = F, sep = "\t", row.names = F)
  return(sigs_detected)
}

get_hrdetect_score <- function(data, std_data){
  intercept <- std_data %>%
    filter(feature == "intercept") %>%
    pluck("weight")
  
  data <- data %>%
    left_join(std_data) %>%
    mutate(norm_value = log(value + 1),
           norm_value = (norm_value - mean)/sd) %>%
    mutate(weighted_value = norm_value*weight) %>%
    group_by(Tumor_Sample) %>%
    summarise(sum_values = sum(weighted_value)) %>%
    mutate(hrdetect_score = 1/(1 +exp(-(intercept + sum_values)))) %>%
    select(-sum_values)
  
  return(data)
}


```


```{bash create_dirs}

mkdir -p data/raw/ascat
mkdir -p data/raw/indels
mkdir -p data/raw/maf
mkdir -p data/raw/microhom
mkdir -p data/raw/qsv
mkdir -p data/processed/scarHRD
mkdir -p figures
mkdir -p tables

```



### Load somatic report

First need to provide the somatic report that will indicate which files to include in the analysis. This report should be placed in 'data/supporting/' directory.  


```{r meta, message=FALSE}

#Analysis path - by default is the path where Rmd script is located
dir <- getwd()

# Replace example file name with somatic report that will be used for this analysis
somatic_report <- "data/supporting/ICGC_SomaticReport.txt"

meta <- read_tsv(somatic_report, show_col_types = FALSE)

# Writing environment variables so that bash can use them
Sys.setenv(somatic_report = somatic_report)
Sys.setenv(dir = dir)
```


### HRdetect prep

#### Process indels 

Preprocessing of indels - using a custom scripts (scripts/microhomology) developed by Eric Zhao - https://github.com/eyzhao/hrdetect-pipeline  


```{bash preprocess_indels}
cat $somatic_report | \
awk -v dir=$dir 'BEGIN {FS="\t"; OFS=""};  NR != 1 {print "ln -sf ",$61,"/*.shc.vcf.gz "dir"/data/raw/indels/"}' > scripts/copy_indels.txt

sh scripts/copy_indels.txt


qsub -v DIR=$dir scripts/run_indels_to_mhom.pbs 


#monitoring for temp file that indicates that the qsub job is done
while [ ! -f ./data/supporting/indels_to_mhom_job_done.txt ]; do sleep 20; done

# remove temp file for next time
rm ./data/supporting/indels_to_mhom_job_done.txt

```

Load up all microhomology data pre-calculated using Eric Zhao's scripts.  

```{r microhom}

microhom <- list.files(path="data/raw/microhom/",
                       full.names = T,
                       pattern = ".microhomology_results.tsv$",
                       recursive = F) %>%
  purrr::set_names(. %>% basename() %>% str_remove(".microhomology_results.tsv")) %>% 
  map_dfr(~read_tsv(.,col_types = cols()), .id = "name") %>%
  mutate(name= str_replace(name, "[.]","_")) %>%
  separate(name, into = c("Tumor_Sample"), 
           extra = "drop", remove = TRUE, sep = "[.]")


```



#### Mutational signature analysis

Define chromosomal order.  

```{r chrom_order}
#Chromosome order
chr_levels=c("1","2","3","4","5","6","7","8","9","10","11",
             "12","13","14","15","16","17","18","19","20","21",
             "22","X")
```

Load MAF files with somatic variants.  
  
```{bash link_mafs}

#cat $somatic_report | awk -v dir=$dir 'BEGIN {FS="\t"; OFS=""};  NR != 1 {print "ln -sf ",$61,"/*.shc.maf.gz #"dir"/data/raw/maf/"}' > scripts/copy_maf.txt

#sh scripts/copy_maf.txt

ln -s /working/genomeinfo/share/analysis/AOCS_-_ICGC_ovarian_cancer_project/MAFs/*.shc.maf.gz data/raw/maf/
```

Run deconstructSigs analysis - mutational signature assignment.  

```{r snv_sigs, message=FALSE}

df_maf_sp <- list.files(path="data/raw/maf/",
            full.names = T,
            pattern = ".shc.maf.gz$",
            recursive = F)  %>%
  map_dfr(~read_tsv(., col_types = cols(.default = "c"),
                    comment="#") %>%
            select(1:62))  %>%
  mutate(Tumor_Sample_Barcode = str_replace(Tumor_Sample_Barcode,":","_")) %>%
  separate(Tumor_Sample_Barcode, into = c("Tumor_Sample"),
           sep=":", extra="drop", remove =FALSE)


snv_count <- df_maf_sp %>%
  filter(Variant_Type == "SNP") %>%
  count(Tumor_Sample)

# All combined
df_maf_sp_SNV <- df_maf_sp %>% 
  filter(Variant_Type == "SNP") %>%
  mutate(alt = if_else(Reference_Allele == Tumor_Seq_Allele1, 
                      Tumor_Seq_Allele2, Tumor_Seq_Allele1)) %>%
  filter(Chromosome %in% chr_levels) %>%
  mutate(chr = paste0("chr",Chromosome)) %>%
  mutate(Start_Position = as.numeric(Start_Position)) %>%
  select(Tumor_Sample, chr, Start_Position, Reference_Allele, alt) %>%
  as.data.frame()

# Convert to mutation matrix
sigs_input <- mut.to.sigs.input(mut.ref = df_maf_sp_SNV, 
                                sample.id = "Tumor_Sample", 
                                chr = "chr", 
                                pos = "Start_Position", 
                                ref = "Reference_Allele", 
                                alt = "alt")


# Read in Cosmic v2 signatures.
cosmic_t <- read_delim("./data/supporting/signatures_probabilities.txt",
                      delim = "\t")  %>%
  select(1:33,-`Substitution Type`,-Trinucleotide) %>%
  arrange(match(`Somatic Mutation Type`,colnames(sigs_input))) %>%
  column_to_rownames(var = "Somatic Mutation Type") %>%
  as.matrix()
 
cosmic_sigs <- as.data.frame(t(cosmic_t)) %>%
  rownames_to_column(var = "Signature") %>%
  mutate(Signature = str_replace(Signature,"Signature ","Sig")) %>%
  column_to_rownames(var = "Signature")


sig_cosmic_default <- run_deconstructsigs(data = sigs_input, 
                                          out_path = "./",
                                          out_name = "cosmicv2_sig",
                                          signatures = cosmic_sigs,
                                          method = "default", 
                                          plot = T)
sig_cosmic_default <- read_table("tables/default_cosmicv2_sig.tsv")

snv_hrdetect_sigs <- sig_cosmic_default %>%
  dplyr::rename(Tumor_Sample=Sample) %>%
  left_join(snv_count) %>%
  mutate_at(vars(starts_with("Sig")),list(~.*n)) %>%
  select(Tumor_Sample, Sig3, Sig8)

```


#### Rearrangement signature analysis

First step - copy qSV outputs to preprocess.  
  
```{bash copy_qsv}

#cat $somatic_report | \
#awk -v dir=$dir 'BEGIN {FS="\t"; OFS=""};  NR != 1 {print "cp #",$73,"/*.somatic.ascat.mere.type.annotated.HighConfidence.dcc "dir"/data/raw/qsv/"}' > scripts/copy_qsv.txt

#sh scripts/copy_qsv.txt


ln -s /working/genomeinfo/share/analysis/AOCS_-_ICGC_ovarian_cancer_project/qsv/highconfidence/* data/raw/qsv/

```


Pre-process qSV outputs for SV signature analysis (script written by Felicity Newell).  
  

```{bash qsv_annot_list}

cat $somatic_report | awk -v dir=$dir 'BEGIN {FS="\t"; OFS=""};  NR != 1 {print  dir"/data/raw/qsv/", $3, ".", $71, ".somatic.ascat.mere.type.annotated.HighConfidence.dcc\t",$9,"_",$16}' | sed 's/analysis://'| sort | uniq > data/supporting/qsv_annot_list.tsv

```
```{r}

qsv_annot_list <- read_tsv("data/supporting/qsv_annot_list.tsv", col_names = F)
rm_list <- c()
for (i in qsv_annot_list$X1){
  if(!file.exists(i)){
    rm_list <- c(rm_list, i)
  }
}

qsv_annot_list <- qsv_annot_list %>%
  filter(!X1 %in% rm_list)

write.table(qsv_annot_list, "data/supporting/qsv_annot_list.tsv", append = F, quote = F, col.names = F, row.names = F, sep = "\t")
```
```{bash}
qsub -v DIR=$dir scripts/run_sv_signature_counts.pbs


#monitoring for temp file that indicates that the qsub job is done
while [ ! -f ./data/supporting/sv_signature_count_job_done.txt ]; do sleep 10; done

# remove temp file for next time
rm ./data/supporting/sv_signature_count_job_done.txt



```

  
Run YAPSA to assign RS (SV) signatures. Starting point matlab input file created by Felicity.  
   
```{r SV_sig_assignment}

sv_counts <- read_tsv("data/processed/sv_signature_matrix.tsv",show_col_types = FALSE)


sv_total_counts <- sv_counts %>%
  select(-Type,-SubType) %>%
  gather(key = "patient_testSampleSubmittedWithID", value = "N") %>%
  group_by(patient_testSampleSubmittedWithID) %>%
  summarise(Total_count = sum(N))


nikzainal_SV_sigs <- read_tsv("data/supporting/nz_sv_sigs.txt",show_col_types = FALSE) %>%
  arrange(factor(subtypes, levels = sv_counts$SubType)) %>%
  column_to_rownames("subtypes")

sv_counts %<>%
   select(-Type) %>%
  column_to_rownames("SubType")

sv_sigs_raw <- LCD(
  in_mutation_catalogue_df = sv_counts,
  in_signatures_df = nikzainal_SV_sigs)

sv_sigs_assigned <- sv_sigs_raw %>%
  rownames_to_column("Sig") %>%
  gather(key= patient_testSampleSubmittedWithID, value = count_raw, - Sig) %>%
  left_join(sv_total_counts)  %>%
  group_by(patient_testSampleSubmittedWithID) %>%
  mutate(Count = count_raw/sum(count_raw)*Total_count) %>%
  mutate(Perc = Count/Total_count*100) %>%
  mutate(Count = if_else(Perc > 10, round(Count), 0)) %>%
  mutate(Perc = if_else(Perc > 10, Perc, 0))  %>%
  #Need to change next time - accidentally used controlSampleSubmittedWithID instead of test for SVs!
  left_join(meta %>% 
              select(patientSubmittedWithID,testSampleSubmittedWithID,patientLabel, testSampleLabel) %>%
              mutate(patient_testSampleSubmittedWithID = paste0(patientSubmittedWithID,"_",testSampleSubmittedWithID))) %>%
  mutate(Tumor_Sample = paste(patientLabel, testSampleLabel, sep ="_")) %>%
  ungroup() %>%
  select(-Total_count, -count_raw, -testSampleSubmittedWithID,-patientLabel,-testSampleLabel,-patient_testSampleSubmittedWithID)


```
ß#### HRD score analysis


Link relevant CNV files (ascat outputs).  

```{bash ln_ascat}



```

Run scarHRD to get HRD scores from ascat CN:

```{r scarHRD, message = FALSE}
# Load all ascat files into one dataframe
CNV <- list.files(path="data/raw/ascat/",
                         pattern = ".copynumber.ascat.tsv$",
                         recursive = F,
                         full.names = T) %>%
  map_dfr(~read_tsv(.,show_col_types = FALSE)) %>%
  dplyr::rename(ascatAnalysis = sample)


CNV_stats <- list.files(path="data/raw/ascat/",
                        pattern = ".summary.stats.tsv$",
                        recursive = F,
                        full.names = T) %>%
  map_dfr(~read_tsv(.,show_col_types = FALSE)) %>%
  dplyr::rename(ascatAnalysis = analysis)

CNV_temp <-CNV %>%
  left_join(CNV_stats) %>%
  mutate(SampleID = ascatAnalysis,
         Chromosome = paste("chr",chromosome,sep=""),
         Start_position = start, End_position = end,
         total_cn = tumour.total.cn, A_cn = tumour.minor.cn, B_cn = tumour.major.cn) %>%
  select(SampleID, Chromosome, Start_position, End_position,
         total_cn, A_cn, B_cn, ploidy) %>%
  distinct()%>%
  group_by(SampleID) %>%
  group_walk(~ {
    write_tsv(.x, paste0("data/processed/scarHRD/",.y$SampleID, ".scarHRD.input.txt"))
    scar_score(paste0("data/processed/scarHRD/",.y$SampleID, ".scarHRD.input.txt"),
               reference = "grch37" ,seqz = FALSE, outputdir = "data/processed/scarHRD/")
    }, keep = TRUE)
  

scarHRD_summary <- list.files(path="data/processed/scarHRD",
                            pattern = "_HRDresults.txt$",
                            full.names = T,
                            recursive = F) %>%
  map_dfr(~read_tsv(.,col_types = cols()))  %>%
  mutate(ascatAnalysis = paste0("analysis:",...1))  %>%
  left_join(meta %>% select(patientLabel, testSampleLabel, ascatAnalysis))  %>%
  mutate(Tumor_Sample = paste(patientLabel, testSampleLabel, sep ="_"))
```


### HRdetect predictions

First, clean up all the inputs for HRdetect input.  

```{r clean_up}

hrdetect_input <- snv_hrdetect_sigs %>%
  full_join(scarHRD_summary %>% select(Tumor_Sample, `HRD-sum`) %>% 
              distinct()) %>%
  full_join(microhom %>% select(Tumor_Sample, deletion_microhomology_proportion)%>%
              distinct()) %>%
  inner_join(sv_sigs_assigned %>%
              select(Tumor_Sample, Count, Sig) %>%
              mutate(Sig = str_replace(Sig, "Rearrangement Signature ","rs")) %>%
               distinct() %>%
              spread(key = Sig, value = Count) %>%
              select(Tumor_Sample, rs3,rs5)) %>%
  select(Tumor_Sample,Sig3,Sig8,
         rs3,rs5,
         `HRD-sum`,deletion_microhomology_proportion) %>%
  drop_na() %>%
  gather(key = feature, value = value, 
         -Tumor_Sample) %>%
  mutate(feature = case_when(feature == "Sig3" ~ "sig3",
                             feature == "Sig8" ~ "sig8",
                             feature == "HRD-sum" ~ "hrd",
                             feature == "deletion_microhomology_proportion" ~ "mhom",
                             TRUE ~ feature)) %>%
  distinct()

```

```{r hrdetect_manual}

hrd_adj_val <- read_tsv("data/supporting/hrdetect_adjustment_values.tsv", show_col_types = FALSE)


hrdetect_output <- get_hrdetect_score(hrdetect_input,hrd_adj_val)

```

```{r summary_output}
hrdetect_output %>%
  left_join(scarHRD_summary %>% select(Tumor_Sample, `HRD-sum`)) %>%
  write_tsv("tables/hrdetect.tsv")

sig_cosmic_default %>%
  dplyr::rename(Tumor_Sample=Sample) %>%
  left_join(snv_count) %>%
  gather(key = Sig, value = Perc, -n, -Tumor_Sample) %>%
  mutate(Count =round(Perc*n)) %>%
  mutate(Perc = Perc*100) %>%
  select(-n) %>%
  filter(Count != 0) %>%
  arrange(Tumor_Sample) %>%
  filter(Tumor_Sample %in% hrdetect_output$Tumor_Sample) %>%
  write_tsv("tables/snv_sigs.tsv")

sv_sigs_assigned %>%
  arrange(Tumor_Sample) %>%
  filter(Perc > 0 ) %>%
  write_tsv("tables/sv_sigs.tsv")  
  
```

### R Session

```{r session_info}

devtools::session_info()

```