---
title: "TCGA_OV_validation"
author: "Lijun xU"
date: "2023-11-07"
output: html_document
---

```{r setup, include=FALSE}
library("data.table")
library("purrr")
library("tidyverse")
library("scales")
library("ComplexHeatmap")
library("rstatix")
library("ggrepel")
library("ggplot2")
library("gridExtra")
library("readxl")
library("circlize")
library("ggbeeswarm")
library("ggpubr")
library("edgeR")
library("org.Hs.eg.db")

source("scripts/beta_visualisation_support.R")
source("scripts/mRNA_edgeR_support.R")

```

#palette
```{r}
methylation_color = c("BRCA1 & RAD51C" = "#46351D","BRCA1" = "#93a667", "RAD51C" = "#7BB2D9", "No" = "grey")
methylation_color_extended= c("BRCA1" = "#93a667", "RAD51C" = "#7BB2D9", "No" = "grey",
                              "BRCA2" = "#FAA613", "PALB2" = "#D90368", "BRIP1" = "#4F4789")
methylation_color_ht = c("BRCA1 & RAD51C" = "#46351D","BRCA1" = "#93a667", "RAD51C" = "#7BB2D9", "No" = "white")
methylation_color_brca1 = c("Yes" = "#93a667","No" = alpha("grey", 0.6))
methylation_color_rad51c = c("Yes" = "#7BB2D9","No" = alpha("grey", 0.6))
col = list(alt = c("MUT" = "#EE8A8A", "Conseq" = "#616161", "NA" = "white"),
           Methylation = methylation_color,
          CIMP = c("CIMP-" = "#D6F8D6", "CIMP+" = "#7FC6A4", "CIMPi" = "#B0C7BD", "NA" = "white"),
           `CpG Methylation` = c("Low" = "#D6F8D6", "High" = "#7FC6A4", "Intermediate" = "#B0C7BD", "NA" = "white"),
           `STAD Subtype` = c("EBV" = "#CB6434", "CIN" = "#E09A75", "MSI" = "#F5CFB5", "NA" = "white"),
           `LGG Subtype` = c("IDHmut codel" = "#BB9F23", "IDHmut non-codel" = "#DDCE74", "IDHwt" = "#FFFCC5", "NA" = "white"),
           `UCEC Subtype` = c("CN HIGH" = "#3F72AF", "MSI" = "#90caf9", "NA" = "white"),
           `TGCT Subtype` = c("Mixed Germ Cell Tumor" = "#DC778D", "Yolk Sac Tumor" = "#E4B1BC","NA" = "white"),
           `BC Subtype` = c("LumA" = "#AAA1C8", "LumB" = "#D5C6E0", "Basal" = "#192A51", "Normal" ="#F5E6E8" , "Her2" = "#967AA1", "NA" = "white"),
           Meth_lvl = c("HM"= "#F9A620","LM"= "#F8DD83","NA" = "white"))

col_sig = list(Methylation = c("BRCA1 & RAD51C" = "#46351D","BRCA1" = "#93a667", "RAD51C" = "#7BB2D9", "No" = "white"))

methylation_lvl <- c("BRCA1 & RAD51C","BRCA1", "RAD51C", "No")
project_lvl <- c("OV", "TGCT", "STAD","LGG","BC", "UCEC", "LIHC", "COAD", "LUSC", "HNSC", "THCA")
hr_gene_list <- c("BRCA1", "BRCA2", "RAD51C", "PALB2", "BRIP1", "RAD51D")

RAD51C_het_hom <- c("Homozygous RAD51Cme" =  "#165E91", "Heterozygous RAD51Cme" = "#7BB2DA", "No Methylation" =  "grey")

HRD_color <- colorRamp2(breaks = c(0, 42, 100), colors = c("#AEB6E5", "white", "#A93154"))
prom_meth_color <- colorRamp2(breaks = c(0, 0.5, 1), colors = c("#5D3A9B", "white", "#E66100"))
avg_corr_meth_color <- colorRamp2(breaks = c(0, 0.5, 1), colors = c("#0C7BDC", "white", "#FFC20A"))
signature_color <- colorRamp2(breaks = c(0,0.1, 1), colors = c("white","#F1F1F1", "#26A63A"))
Purity = colorRamp2(breaks = c(0, 0.5, 1), colors = c("white", "grey", "black"))
 `CN` =colorRamp2(breaks = c(0, 2, 8), colors = c("#4D7C93", "white", "#CC5C3E"))
 
 alter_fun = list(
    background = function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill = "#EBEBEB", col = NA))
    },
    `NA` = function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill = "white", col = NA))
    },
    `Nonsense` = function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill =  "#8B7774", col = NA))
    },
    `Frameshift` = function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill =  "#C17767", col = NA))
    },
    `Splice Site` = function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill =  "#E4924E", col = NA))
    },
   `In Frame`= function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill =  "#EDD382", col = NA))
    },
   `Missense`= function(x, y, w, h) {
        grid.rect(x, y, w-unit(2, "pt"), h-unit(2, "pt"),
            gp = gpar(fill =  "#719DB1", col = NA))
    },
    `ClinVar P/LP` = function(x, y, w, h) {
        grid.roundrect(x, y, w-unit(2, "pt"), h*0.1,
            gp = gpar(fill = "#616161", col = NA))
    }
)

```
##purity
```{r}
absolute_purity <- fread("data/processed/TCGA_ABSOLUTE_purity.txt")%>%
  mutate(Sample.ID = array) %>%
  dplyr::rename(absolute_purity = purity) %>%
  dplyr::select(Sample.ID, absolute_purity)
```
# Preprossing (Only need to run once)
## make sub methylation matrix
```{r}
source("scripts/beta_visualisation_support.R")
genecodefile =  "data/supporting/HM27.hg38.manifest.gencode.v36.tsv.gz"
gene_promoter_info = "data/supporting/EPDnew_human_ver006_hg38.tsv"
HR_gene_list = c("BRCA1","BRCA2","RAD51C","RAD51D","BRIP1","PALB2","CHEK2", "XRCC3", "RAD51B")


beta_matrix <- Sys.glob("data/side_project/TCGA_OV_norm_beta.csv") %>%
  map_dfr(~ as.data.frame(fread(.x))) %>%
  mutate(CpG = V1)%>%
  column_to_rownames("V1")

# load sample
sample_sheet <- as.data.frame(fread(Sys.glob(paste0("data/raw/2023*_*_meth/IDAT/*_pd.csv")), header = T))
# load pvalue
#pvalue <- as.data.frame(fread(Sys.glob(paste0(directory, "/data/processed/methylation/", study_name, "_beta_pvalue.csv")), header = F)) %>%
#  'colnames<-'(c("Sample_Name", "pvalue"))
# subset beta
beta_sub <- HR_gene_list %>%
                   map_df(~(data.frame(gene = .x,
                                       CpG_probe_by_gene(., genecodefile)))) %>%
                   left_join(beta_matrix, by="CpG") %>%
                   # create columns with CpG name and corresponding values
                   pivot_longer(cols = -c("CpG", "chr_start", "chr_end", "gene", "probe_strand", "distToTSS", "CpG_context"), names_to = "Sample_Name", values_to = "beta") %>%
                  mutate(Sample_Name = if_else(startsWith(Sample_Name, "GSM"),substr(Sample_Name, 1, 10) , Sample_Name)) %>%
                   # add sample info
                   left_join(sample_sheet, by="Sample_Name")

# annotate subset beta file with CpG genome context around HR gene
gene_promo <- read.table(gene_promoter_info, header= T)

##add a column with pre-defined CpG, these probes were manually assigned from variable assessment
beta_sub <-beta_sub %>%
  mutate(PredefinedCpG = case_when(
    chr_start >= 43125041 & chr_end <= 43125714 ~ "BRCA1",
    chr_start >= 58692529 & chr_end <= 58693064 ~ "RAD51C"
  ))

##file only contains putative promoter starting, and ending location
gene_promo <- gene_promo %>%
  mutate(EPD_promoter = rownames(gene_promo),
         Start_200 = Start - 200,
         End_200 = End + 200)


##setting data.table for unequal joining
setDT(beta_sub)
setDT(gene_promo)

##unequal joining for gene annotation
beta_sub[gene_promo, on =.(chr_start > Start, chr_end < End), EPD_promo := EPD_promoter]
beta_sub[gene_promo, on =.(chr_start > Start_200, chr_end < End_200), gene_promo_200 := Gene]

#save beta file to the processed data
fwrite(as.data.table(beta_sub, keep.rownames = ""),"data/side_project/TCGA_OV_beta_subset_promo.csv", row.names = F, sep = "\t")

```
## mRNA
```{r}
study_name = "TCGA_OV"
study_dir = "data/raw/20230711_TCGA_OV_meth/mRNA"
output_dir =  "data/side_project"
plot_output_dir = "data/side_project/OV_mRNA"

#gene annotation
gene_annotation <- fread(paste0(working_dir, "data/supporting/biomaRt_ensg_annotation.csv"), drop =1, header=T)


# prepare count, sample and gene matrix
mRNA_counts <- Sys.glob(paste0(study_dir, "/*gene*ounts*"))%>%
  map_dfr(~as.data.frame(data.table::fread(.x, skip = 6, header = F) %>%
                           mutate(filename = basename(.x)))) %>%
  dplyr::select(gene_ID = V1, gene_name = V2, unstranded_counts = V4, filename)

#sum mRNA count for each file to filter duplicates
sum_counts <- mRNA_counts %>%
  ungroup() %>%
  group_by(filename) %>%
  summarise(sum_counts=sum(unstranded_counts))

gene_csv <- mRNA_counts %>%
  dplyr::select(gene_ID, gene_name) %>%
  mutate(ensembl_gene_id = substr(gene_ID, 1, 15)) %>%
  left_join(gene_annotation, by = "ensembl_gene_id") %>%
  distinct() 

mRNA_counts <- mRNA_counts %>%
  pivot_wider(names_from = filename, values_from = unstranded_counts) %>%
  dplyr::select(-gene_name) %>%
  column_to_rownames("gene_ID")

sample <- as.data.frame(fread(paste0(study_dir, "/TCGA_OV_mRNA_sample.tsv")))
#fix column names
colnames(sample) <- make.names(colnames(sample))

#add methylation information for design matrix
sample <- sample %>%
  dplyr::select(filename=File.Name, Sample.ID, Sample.Type)

#filter for higher sum counts among duplicates
filtered_filename <- sum_counts %>%
  left_join(sample, by = c("filename")) %>%
  ungroup() %>%
  group_by(Sample.ID) %>%
  filter(sum_counts == max(sum_counts)) %>%
  ungroup() %>%
  dplyr::select(filename)

sample_list <- sample %>%
  filter(filename %in% unlist(filtered_filename))  #design matrix have BRCA1 and RAD51C_M information combined

#have column matched between mRNA counts and sample information
mRNA_counts <- mRNA_counts[,sample_list$filename]

#form star_matrix
star_matrix <- DGEList(counts = mRNA_counts, group = factor(sample_list$Sample.Type), samples = sample_list)


#QC
##plot raw expression distribution in 50 random sample
ran_sample <- sample(dim(star_matrix)[2], 50)
pdf(paste0(plot_output_dir, "/QC/", study_name, "_raw_distribution50.pdf"))
boxplot(cpm(star_matrix[,ran_sample], log =T), use.cols = T)     
dev.off()

##filter out low count gene
keep <- filterByExpr(star_matrix)
star_matrix <- star_matrix[keep,, keep.lib.sizes=FALSE]
star_matrix <- calcNormFactors(star_matrix)

##re-plot expression distribution
pdf(paste0(plot_output_dir, "/QC/", study_name, "_filter_norm_distribution50.pdf"))
boxplot(cpm(star_matrix[,ran_sample], log =T), use.cols = T)  
dev.off()


# normalisation
design.mat <- model.matrix(~star_matrix$samples$group)
star_matrix <- estimateDisp(star_matrix, design.mat, robust= T)
norm_counts <- cpm(star_matrix, log=TRUE)
fwrite(as.data.table(norm_counts, keep.rownames = ""), paste0(output_dir, "/", study_name, "_mRNA_edgeR.csv"))

##plot BCV
pdf(paste0(plot_output_dir, "/QC/", study_name,"_plotBCV.pdf"))
plotBCV(star_matrix)
dev.off()
```
## copynumber
```{bash}
grep -Er "BRCA1|RAD51C" data/raw/20230711_TCGA_OV_meth/copynumber | grep -v "BRCA1P1" |  sed 's/^\.\///g' > data/side_project/TCGA_OV_BRCA1_RAD51C_cnv.tsv;
```
# Analysis
## load methylation and clinical
```{r}
beta_sub_meth <- BRCA1_RAD51C_meth_annotat(beta_sub %>%
                                            distinct()) %>%
  mutate(Project = "TCGA_OV")

beta_sub_meth %>%
  dplyr::select(Sample.ID, BRCA1_M, Sample_Group) %>%
  distinct() %>%
  filter(Sample_Group == "Primary Tumor") %>%
  mutate(Sample.ID = substr(Sample.ID, 1, 15)) %>%
  left_join(absolute_purity) %>%
  group_by(BRCA1_M) %>%
  count()


beta_sub_meth %>%
  filter(Sample_Group == "Primary Tumor") %>%
  dplyr::select(Project, Sample.ID, BRCA1_M, Sample_Group) %>%
  distinct() %>%
   mutate(CaseID = substr(start = 1,stop = 12, Sample.ID) ) %>%
   left_join( map_dfr(Sys.glob("data/raw/2023*/*clinical.tsv"), ~ as.data.frame(fread(.x, header = T, na.strings = "'--", fill = T,  colClasses = "character")))%>%
               mutate(age = as.numeric(age_at_index)) %>%
                dplyr::select(stage = ajcc_pathologic_stage, gender, race, age, CaseID=case_submitter_id))%>%
  mutate(stage = if_else(is.na(stage), "NA", stage)) %>%
    distinct()-> demograph_table

```
## methylation visualisation
```{r}
suppressMessages(suppressWarnings(require(ggnewscale)))
beta_plot <- beta_plot_context_annotat(beta_sub %>%
                                         filter(gene == "BRCA1")%>%
                                         mutate(gene_promo_200 = factor(gene_promo_200, levels = c("BRCA1")),
                                                PredefinedCpG = factor(PredefinedCpG, levels = c("BRCA1")),
                                                EPD_promo = factor(EPD_promo, levels = c("BRCA1")))
                                        , study_name)
beta_plot_saver(beta_plot, group = length(unique(beta_sub$Sample_Group)), output_dir = paste0(directory,"/plots/"))

```
## OV age validation TCGA
```{r}
age_DF <- demograph_table%>%
  mutate(stage = gsub("Stage ", "", stage),
         BRCA1_M= factor(BRCA1_M, levels= c("Yes", "No")))%>% 
  group_by(BRCA1_M) %>%
  mutate(Methylation_count = paste0(" (N=", n_distinct(Sample.ID), ")")) %>%
  ungroup() 


age_stat <- compare_means(age ~ BRCA1_M, 
                          data = age_DF,
                          ref.group = "No")

age_DF %>%
  ggplot(aes(y = age, x = BRCA1_M))+
 geom_boxplot(aes(fill = BRCA1_M))+
  theme_bw()+
  ylab("Age at Diagnosis")+
  theme(legend.position = "bottom", axis.title.x = element_blank(),strip.background = element_rect(fill="white"))+
  ylim(0, 105)+
  stat_pvalue_manual(age_stat%>%
 mutate(y.position = 95), label = "p.adj")+
  geom_text(data =. %>%
              dplyr::select(BRCA1_M, Methylation_count) %>%
              distinct(), mapping = aes(y = 0, label = Methylation_count))+
  scale_fill_manual(values = c("Yes" = "#93a667", "No"= "grey"), "BRCA1 Methylation")+
  theme(legend.position = "bottom")
ggsave("plots/Supplementary/Sup3b_TCGA_OV_age.pdf", width = 4, height = 4)

write.csv(age_DF %>%
            dplyr::select(Project, Sample.ID, meBRCA1 = BRCA1_M, Sample_Group, age) %>%
            distinct(),
          "data/source_data/supplementary_figure_3b.csv", quote = F, row.names = F)

```