#!/bin/bash
################################################################################
################################################################################
#PBS -M lijun.xu@qimrberghofer.edu.au
#PBS -N VEP
#PBS -S /bin/bash
#PBS -j n
#PBS -l walltime=10:00:00
#PBS -l ncpus=2
#PBS -l mem=30gb
#PBS -m ae
#PBS -W umask=0007
#PBS -v working_dir
################################################################################
################################################################################
set -e

############
# prep VCF #
############
cd ${working_dir}
module load R/4.2.0
Rscript ${working_dir}/scripts/vep_variants_annotation_depmap.R  ${working_dir}

#####scripts made by Aimee Davidson#####
###########
# load VEP #
###########
module load testing/1.0 vep/102
input_file="${working_dir}/data/processed/VCF/depmap_variants_hg38.vcf"
output_filestem="${working_dir}/data/processed/VCF/$(basename "${input_file}" "_depmap.vcf")"
env
cmd=(vep)
cmd+=(--input_file "${input_file}")
cmd+=(--everything)
cmd+=(--species homo_sapiens)
cmd+=(--assembly GRCh38)
cmd+=(--output_file "${output_filestem}.vep102.tsv")
cmd+=(--stats_file "${output_filestem}.vep102.html")
cmd+=(--warning_file stderr)
cmd+=(--cache)
cmd+=(--dir_cache /reference/software/vep/102)
cmd+=(--dir_plugins "${HOME}/.vep/VEP_plugins")
cmd+=(--no_progress)
cmd+=(--offline)
#cmd+=(--plugin "ExAC,/reference/data/ExAC/release1/subsets/ExAC_nonTCGA.r1.sites.vep.vcf.gz,AC,AN")
#cmd+=(--plugin "ExACpLI")
#cmd+=(--plugin "CADD,/reference/software/CADD/CADD-1.3/whole_genome_SNVs.tsv.gz,/reference/software/CADD/CADD-1.3/InDels.tsv.gz")
#cmd+=(--plugin "dbNSFP,/reference/data/dbNSFP/2.9.3/dbNSFP2.9.3.gz,ALL")
#cmd+=(--plugin "dbscSNV,/reference/data/dbscSNV/1.1/dbscSNV1.1.txt.gz")
#cmd+=(--plugin "MaxEntScan,/reference/data/MaxEntScan/fordownload,SWA,NCSS,verbose")
#cmd+=(--plugin "REVEL,/working/genomeinfo/share/REVEL/revel_all_chromosomes.tsv.gz")
#cmd+=(--plugin "HGVSIntronOffset")
#cmd+=(--plugin "LRG_RefSeqGene,/working/genomeinfo/share/LRG_RefSeqGene/LRG_RefSeqGene")
cmd+=(--plugin "gnomAD,/reference/data/gnomAD/gnomad-public/r2.1.1/vcf/genomes,ALL")
#cmd+=(--plugin "gnomADc,/reference/data/gnomAD/gnomad-public/r2.0.2/coverage/exomes") # keeping old version for cov because plugin doesnt work for r.2.1.1
#cmd+=(--custom /reference/data/clinvar/vcf_GRCh37/clinvar_20221001.vcf.gz,ClinVar,vcf,exact,0,ALLELEID,CLNDN,CLNDNINCL,CLNDISDB,CLNDISDBINCL,CLNHGVS,CLNREVSTAT,CLNSIG,CLNSIGCONF,CLNSIGINCL,CLNVC,CLNVCSO,CLNVI,DBVARID,GENEINFO,MC,ORIGIN,RS,SSR,CLNSUBA,CLNSIGA,CLNDATEA,CLNDATESUBA,CLNREVSTATA,CLNORA,CLNSCVA,CLNDNA,CLNCOMA)
cmd+=(--custom /reference/data/clinvar/vcf_GRCh38/clinvar_20231007.vcf.gz,ClinVar,vcf,exact,0,ALLELEID,CLNDN,CLNDNINCL,CLNDISDB,CLNDISDBINCL,CLNHGVS,CLNREVSTAT,CLNSIG,CLNSIGCONF,CLNSIGINCL,CLNVC,CLNVCSO,CLNVI,DBVARID,GENEINFO,MC,ORIGIN,RS,SSR,CLNSUBA,CLNSIGA,CLNDATEA,CLNDATESUBA,CLNREVSTATA,CLNORA,CLNSCVA,CLNDNA,CLNCOMA)
cmd+=(--tab)
cmd+=(--dont_skip)
cmd+=(--allow_non_variant)
#cmd+=(--fasta /reference/genomes/GRCh37_ICGC_standard_v2/indexes/SAMTOOLS_1.4/GRCh37_ICGC_standard_v2.fa)
cmd+=(--fasta /reference/genomes/GRCh38_no_alt_analysis_set/indexes/SAMTOOLS_1.2/GCA_000001405.15_GRCh38_no_alt_analysis_set.fa)
cmd+=(--force_overwrite)
cmd+=(--merged)
cmd+=(--no_escape)
cmd+=(--fork 1)
###########
# Run VEP #
###########
stdout="${output_filestem}.vep102.out"
stderr="${output_filestem}.vep102.err"
echo "${cmd[@]}" >> "${output_filestem}.vep102.log"
"${cmd[@]}" 1> "${stdout}" 2> "${stderr}"
